document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java

```diff
@@ -86,38 +105,75 @@ private String getAccountLink(Accounts accounts, AccountType accountsType) {
-
     /**
      * Creates the {@link DocumentInfoResponse} object
      * @param transaction transaction data
      * @param accountType account type
-     * @param abridgedAccountData abridged account data
+     * @param accountData The account data
+     * @param mimeType
      * @return {@link DocumentInfoResponse} object
      */
-    private DocumentInfoResponse createResponse(Transaction transaction, AccountType accountType, AbridgedAccountsApi abridgedAccountData) {
+    private <T> DocumentInfoResponse createResponse(Transaction transaction, AccountType accountType,
+                                                    T accountData, String mimeType) throws ParseException {
         DocumentInfoResponse documentInfoResponse = new DocumentInfoResponse();
-        documentInfoResponse.setData(createDocumentInfoResponseData(transaction, abridgedAccountData));
+        documentInfoResponse.setData(createDocumentInfoResponseData(transaction, accountData, accountType));
         documentInfoResponse.setAssetId(accountType.getAssetId());
         documentInfoResponse.setTemplateName(accountType.getTemplateName());
         documentInfoResponse.setPath(createPathString(accountType));
+
+        Map<String, String> descriptionValues = new HashMap<>();
+        descriptionValues.put("period_end_on", RESPONSE_DISPLAY_DATE_FORMAT.format(ISO_DATE_FORMAT.
+                parse(getCurrentPeriodEndOn(accountData))));
+
+        documentInfoResponse.setDescriptionValues(descriptionValues);
+        documentInfoResponse.setDescriptionIdentifier(accountType.getEnumerationKey());
+        documentInfoResponse.setContentType(mimeType);
         return documentInfoResponse;
     }
 
     private String createPathString(AccountType accountType) {
         return String.format("/%s/%s", accountType.getAssetId(), accountType.getUniqueFileName());
     }
+
+    /**
+     * Creates the 'data' string in {@link DocumentInfoResponse}.
+     * @param transaction the transaction data
+     * @param accountData the abridged accounts data
+     * @param accountType the type of account
+     * @return data string in {@link DocumentInfoResponse}
+     */
+    private <T> String createDocumentInfoResponseData(Transaction transaction, T accountData, AccountType accountType) {
+        String accountTypeName = accountType.getResourceKey();
+        JSONObject accountJSON = new JSONObject(accountData);
+        JSONObject account = new JSONObject();
+        account.put(accountTypeName, accountJSON);
+        account.put("company_name", transaction.getCompanyName());
+        account.put("company_number", transaction.getCompanyNumber());
+        return account.toString();
+    }
+
+    /**
+     * Get the period end Date from the account data
+     *
+     * @param accountData the account data
+     * @return periodEndDate The formatted periodEndDate
+     * @throws ParseException
+     */
+    private <T> String getCurrentPeriodEndOn(T accountData) throws ParseException {
```

Comment:
Can this return a date object? The period end on is being formatted twice event though it's only called from one place.

> return ISO_DATE_AND_TIME_FORMAT.format(ISO_DATE_FORMAT.parse(date));

> descriptionValues.put("period_end_on", RESPONSE_DISPLAY_DATE_FORMAT.format(ISO_DATE_FORMAT.
>                parse(getCurrentPeriodEndOn(accountData))));